### PR TITLE
Reduced log level to debug for image pull

### PR DESCRIPTION
**Pull request by alena1108**
https://github.com/rancher/rke/issues/1684

### CODE REVIEW

File: docker/docker.go [hunk 1]
```diff
@@ -217,23 +217,23 @@ func IsContainerRunning(ctx context.Context, dClient *client.Client, hostname st
...
 	var err error
 	for i := 1; i <= RetryCount; i++ {
 		logrus.Infof("Checking if image [%s] exists on host [%s], try #%d", containerImage, hostname, i)
 		_, _, err = dClient.ImageInspectWithRaw(ctx, containerImage)
 		if err != nil {
 			if client.IsErrNotFound(err) {
-				logrus.Infof("Image [%s] does not exist on host [%s]: %v", containerImage, hostname, err)
-				return false, nil
+				logrus.Debugf("Image [%s] does not exist on host [%s]: %v", containerImage, hostname, err)
+				return err
```
Comment (superseb) [inline docker/docker.go]: This is not an error situation, returning false, nil is correct as the function is `localImageExists` and the error from the Docker daemon is not found, meaning it does not exist and should return false.

File: docker/docker.go [hunk 2]
```diff
@@ -217,23 +217,23 @@ func IsContainerRunning(ctx context.Context, dClient *client.Client, hostname st
...
 			}
-			logrus.Warnf("Error checking if image [%s] exists on host [%s]: %v", containerImage, hostname, err)
+			logrus.Debugf("Error checking if image [%s] exists on host [%s]: %v", containerImage, hostname, err)
 			continue
 		}
-		logrus.Infof("Image [%s] exists on host [%s]", containerImage, hostname)
-		return true, nil
+		logrus.Debugf("Image [%s] exists on host [%s]", containerImage, hostname)
```
Comment (superseb) [inline docker/docker.go]: This removes the feedback for `logrus.Infof("Checking if image [%s] exists on host [%s], try #%d", containerImage, hostname, i)`

**Comment by alena1108** [comment]
@superseb 

> What is the reason to move from a function that was scoped to checking image locally to splitting it out to just returning error and doing another check (errnotfound in pullimage) in another function?

I think that notFound error is descriptive enough, and introducing one extra param - bool - to indicate whether the image exists, is redundant. Think of k8s getObject call - it returns an error if the object can't be found, and then it's up to the client to check for the error kind. It's also up to the client to decide on how to deal with NotFound type of error. I'm open for the argument :)

File: docker/docker.go (hunk 2 above)
Comment (alena1108) [reply docker/docker.go]: I can add the info back for this one, no problem

File: docker/docker.go [hunk 3]
```diff
@@ -217,23 +217,23 @@ func IsContainerRunning(ctx context.Context, dClient *client.Client, hostname st
...
-			logrus.Warnf("Error checking if image [%s] exists on host [%s]: %v", containerImage, hostname, err)
+			logrus.Debugf("Error checking if image [%s] exists on host [%s]: %v", containerImage, hostname, err)
 			continue
 		}
 		logrus.Infof("Image [%s] exists on host [%s]", containerImage, hostname)
-		return true, nil
+		return nil
 	}
-	return false, fmt.Errorf("Error checking if image [%s] exists on host [%s]: %v", containerImage, hostname, err)
+	return fmt.Errorf("error checking if image [%s] exists on host [%s]: %v", containerImage, hostname, err)
```
Comment (superseb) [inline docker/docker.go]: Why is this `error` and the rest is `Error`?

**Comment by superseb** [comment]
The current changes don't line up with this comment: https://github.com/rancher/rke/issues/1684#issuecomment-539428464

I think we should either move them all to debug, or have feedback on the actions and just strip the error as the issue was that Docker reported it as error while we logged it as info

File: docker/docker.go (hunk 3 above)
Comment (alena1108) [reply docker/docker.go]: @superseb we should fix the rest. It should align with the go best practices where the error should be in lower case

**Comment by alena1108** [comment]
@superseb 

>I think we should either move them all to debug, or have feedback on the actions and just strip the error as the issue was that Docker reported it as error while we logged it as info

reduced level to debug in localImageExists, https://github.com/rancher/rke/pull/1736/files#diff-b715a530d535012e418a564a3eafedb9R223

**Comment by superseb** [comment]
LGTM

**Pull request merged**